packages/lib/src/core/types.ts

```diff
@@ -47,6 +47,14 @@ export interface ICore {
 
 export type AdyenEnvironment = 'test' | 'live' | 'live-us' | 'live-au' | 'live-apse' | 'live-in' | string;
 
+export type onSubmitGenericType<T> = (
+    state: any,
+    element: T,
+    actions: {
+        resolve: (response: CheckoutAdvancedFlowResponse) => void;
+        reject: () => void;
+    }
+) => void;
```

Comment:
I like the idea of trying to narrow the scope of the `onSubmit` function to the respective component, although we should investigate how can we perform this not only for the `onSubmit` but for all other callbacks as well that return an UIElement.

I suggest to start tackling this generic types in another PR.

Reply:
Check if this change needs to be in this PR.